### ISSUE TITLE
Wii U ready courses?

### ISSUE THREAD

**Issue by Qiangong2** [RedDuckss/smm1-course-archive#1]
Are the courses that are archived with this script able to be put directly on a Wii U? Or do they need to be converted first? I'd put this on pretendo, but there isn't an issues tab there

**Comment by jonbarrow** [comment]
Yes these courses should be compatible with SMM1 as they are

**Comment by Qiangong2** [comment]
EDIT: nvm got the deviceID figured out

**Comment by Qiangong2** [comment]
Is it normal for it to fail to get course IDs early on?

**Comment by jonbarrow** [comment]
Yes, the course IDs don't seem to be sequential. They seem to be randomly assigned, so it's perfectly normal for it to fail early on. Given that course uploading is disabled now though and no new course IDs can be made, it's possible to just make a list of all possible course IDs and only loop over those ones. I just don't have such a list so the script opts to loop over every possible value regardless of if it exists or not

**Comment by Qiangong2** [comment]
I'm up to 800 now and still none have been found weirdly enough
I wonder if the IDs are all 4 digits or higher

EDIT: which course number was the first hit for you?

**Comment by jonbarrow** [comment]
I never reached one naturally using the loop implemented here, that's purely for the archival purposes. I just used known course IDs when testing since I don't need to download them all

**Comment by Qiangong2** [comment]
Ah, I'm trying to download all courses. Guess I'll just keep running it until I hit one

**Comment by Qiangong2** [comment]
Should it be saying this when it doesn't find one?
```
INFO:nintendo.nex.datastore_smm:DataStoreClientSMM.prepare_get_object()
Failed to get course 6563
```
Still haven't found any btw, does my serial number need the FW in it? EDIT: Yes you need it

**Comment by Qiangong2** [comment]
I left it overnight and it never found any courses. Why is that? Is the script broken now?

**Comment by jonbarrow** [comment]
The script isn't broken, there's just tens of millions of courses. I doubt the course IDs start anywhere near the lower thousands. If I knew the lowest real course ID I would have the script start there. There's a testing ID in the script that I commented out, you can verify the script works by uncommenting it

**Comment by Qiangong2** [comment]
Ohh, so line 45 doesn't change where it starts.

What line would I change to change the start? That way I can work in increments over a couple weeks

**Comment by jonbarrow** [comment]
What do you mean "change where it starts"

**Comment by Qiangong2** [comment]
Instead of start at course 0 after every stop, start it at course 9500, or course 20000, etc.

**Comment by jonbarrow** [comment]
According to JAA over on IRC:

```
<@JAA> Highest existing ID is E97E-0000-0425-6405
<@JAA> Also, lowest ID from the thumbnails is 03E7-0000-000F-6380
```

They found these numbers after scraping all the thumbnails from the SMM1 bookmark site before it went down, though these numbers might not be 100% accurate

**Comment by Qiangong2** [comment]
Alright, thanks. I'm using a minimal cloud instance now to just let it run. Hopefully this will establish the absolute bottom end and we can work from there

**Comment by KurtStineAI2** [comment]
I got to 3 million and still got nothing. Even the testing ID doesn't work anymore. Is that the case for you as well?

**Comment by Qiangong2** [comment]
I reach 3 million and still nothing. Now even the testing ID isn't working. Is the same thing happening to you?
